src/Compiler/Symbols/Exprs.fs

```diff
@@ -1264,9 +1264,12 @@ module FSharpExprConvert =
         match x with 
         | TDSwitch(inpExpr, csl, dfltOpt, m) -> 
             let acc = 
-                match dfltOpt with 
-                | Some d -> ConvDecisionTreePrim cenv env dtreeRetTy d 
-                | None -> wfail( "FSharp.Compiler.Service cannot yet return this kind of pattern match", m)
+                match dfltOpt, dtreeRetTy with 
```

Comment:
Most of the times, we want to stripTypeEqns on a ttype - to arease aliases and units of measures.
What about here?

Reply:
Thanks, makes sense here, too.